sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -64,6 +66,7 @@
   /**
    * Returns the default {@code Coder} to use for the data read from this source.
    */
+  @Nullable
```

Comment:
1. Yes, I think you should just add the `throws` clause in those cases.
2. You're right, this is a breaking change. However, the Source class itself is marked `@Experimental`. Arguably it shouldn't be, but... it is. In general, still, the threshold for changing the Source API should be very high, but this particular change would affect only users who are _calling_ Source.getDefaultOutputCoder (it would not affect users who are merely _implementing_ it - adding a throws to a base class method signature is a compatible change for subclasses), which I expect is an extremely small subset of advanced users. I think the benefit of this change outweighs this cost.